O365/connection.py

```diff
@@ -978,13 +970,14 @@ def _check_delay(self) -> None:
         self._previous_request_at = time.time()
 
     def _internal_request(
-        self, session_obj: Session, url: str, method: str, **kwargs
+        self, session_obj: Session, url: str, method: str, ignore401: bool, **kwargs
     ) -> Response:
         """Internal handling of requests. Handles Exceptions.
 
         :param session_obj: a requests Session instance.
         :param str url: url to send request to
         :param str method: type of request (get/put/post/patch/delete)
+        :param bool ignore401: indicates whether to ignore 401 error or not
```

Comment:
I think this docstring should indicate the real purpose of this param. This is whether or not try to interpret a 401 as a token expired error.

Reply:
Added some extra test on latest commit.